packages/runtime/container-runtime/src/dataStores.ts

```diff
@@ -448,6 +448,8 @@ export class DataStores implements IDisposable {
 	) {
 		const envelope = message.contents as IEnvelope;
 		const transformed = { ...message, contents: envelope.contents };
+		const request = { url: envelope.address };
+		this.validateNotDeleted(envelope.address, request);
```

Comment:
I am wondering if we should incorporate the delete logic into DataStoreContexts (this.context here) so that if get is called on deleted context, we throw an error there. Otherwise, we may miss some cases that fail because the data store was deleted.
I see a lot of calls to this.context.get in this file and they will all fail if data store was deleted.

Reply:
Good point, probably